doit/dependency.py

```diff
@@ -361,6 +338,96 @@ def remove_all(self):
         self._conn.execute('delete from doit')
 
 
+class BaseChecker(object):
```

Comment:
The name `checker` is too generic. Can we use something like `FileChangedChecker` ? It is ugly and long but much more clear.